### PR TITLE
Enhance styling in [NetworkImage] to be easy and flexible

### PR DESCRIPTION
**Pull request by yujong-lee**
## Description

1. Remove `absolute` in `defaultImage` style.
It was needed in past, but it is now useless and result in unexpected behavior.

2. Add `maxHeight` to `defaultImage` style.
Since `height` is `auto` calculated base on `ratio` and `width: 50%`, `height` can be bigger than boundary. This is not expected behavior, so this should be restricted with `maxHeight`.

3. Add `fallback` in `styles`. Because it is more flexible. Also, it is more predictable for user. Because `Networkimage` has 3 state of image. `image(final image to show)`, `loading` and `fallback`, so it is expected to have 3 properties in `styles` for each of them.

Note that `fallback Image` has `style=[defaultImage, image, fallback]`. This is because it is natural that `image` and `fallbackImage` have same style(such as size), but there can be some edge case. (If `borderRadius` is applied to `image` and `fallbackImage` is transparent logo. This case, `fallback = {{borderRadius : undefined}}` might be needed.)

## Test Plan
none.

## Related Issues
none.

## Tests
none.

## Checklist
- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] Run `yarn test:all` and make sure nothing fails. You can run `yarn test -u` to update snapshots if needed.
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Review by DevYuns — CHANGES_REQUESTED** [review]

File: main/NetworkImage.tsx [hunk 1]
```diff
@@ -99,7 +101,7 @@ function NetworkImage(props: Props): ReactElement {
 
       {!needLoading && !isValidSource && (
         <Image
-          style={[defaultImage, image]}
+          style={[defaultImage, image, fallback]}
```
[the remark this answers is not in the document]
Comment (yujong-lee) [reply main/NetworkImage.tsx]: That's make sense. Thanks for detail example.

**Comment by yujong-lee** [comment]
@DevYuns  BTW, do you think [this line](https://github.com/dooboolab/dooboo-ui/blob/0f1355289f80202ac14f192e2a656b84da9a60d3/main/NetworkImage.tsx#L81) is needed? It is from your [hotfix](https://github.com/dooboolab/dooboo-ui/commit/6fe275cfb562a832b3c0eae0e67529a04088003b).

**Comment by DevYuns** [comment]
> @DevYuns BTW, do you think [this line](https://github.com/dooboolab/dooboo-ui/blob/0f1355289f80202ac14f192e2a656b84da9a60d3/main/NetworkImage.tsx#L80) is needed? It is from your [hotfix](https://github.com/dooboolab/dooboo-ui/commit/6fe275cfb562a832b3c0eae0e67529a04088003b).

It maybe not a nessary thing since that is default value. It was just to clarify a `flexDirection` for others. anyway, you can remove it if you think it is unnessary code 😄 .

**Review by DevYuns — APPROVED** [review]
LGTM! 👏

**Comment by DevYuns** [comment]
@yujong-lee You should support retina to `logo` image!!

**Comment by yujong-lee** [comment]
@DevYuns  Yeah. It is good for clarify that, but since in this case, `View` will always contain only 1 element for each state. So I think It is better to remove it.

For clarity, I'll merge this and try `retina support` in another PR. 
Thanks for your review.
